src/umpire/Umpire.cpp

```diff
@@ -173,6 +173,19 @@ void mark_event(const std::string& event)
       [&](auto& e) { e.name("event").category(event::category::metadata).arg("name", event).tag("replay", "true"); });
 }
 
+std::size_t get_total_memory_usage()
+{
+  auto& rm = umpire::ResourceManager::getInstance();
+  std::size_t total_memory{0};
+
+  for (auto s : rm.getResourceNames()) {
```

Comment:
Is this limited to the base resources (PINNED, UNIFIED_MEMORY, DEVICE, HOST, etc...)? Or does it include the pools built on top of them? Just want to make sure we're not double counting.

Reply:
Just resources - this is the correct way to get everything (including any alignment adjustments) without double counting.